ThunderRequest/RequestController+Auth.swift

```
@@ -0,0 +1,75 @@
+//
+//  RequestController+OAuth.swift
+//  ThunderRequest
+//
+//  Created by Simon Mitchell on 12/12/2018.
+//  Copyright © 2018 threesidedcube. All rights reserved.
+//
+
+import Foundation
+
+typealias AuthCheckCompletion = (_ authenticated: Bool, _ error: Error?, _ needsQueueing: Bool) -> Void
+
+extension RequestController {
+    
+    /// Checks the authentication status for a given request
+    ///
+    /// - Parameters:
+    ///   - request: The request to check authentication for
+    ///   - completion: Closure callback with result
+    func checkAuthStatusFor(request: Request, completion: @escaping AuthCheckCompletion) {
+        
+        guard let authenticator = authenticator else {
+            completion(true, nil, false)
```

So if there's no authenticator, we're deemed to be 'authenticated'? Thinking that perhaps this should return an error instead, as surely if we're looking to auth we need to have an authenticator and not having one would be an issue? Unless there's some legacy reason for this?

---

This is called for every single request (And only internally), and so if no authenticator is provided we assume that requests don't need to be authenticated :) Would be a bit of a barrier to entry if users had to provide an `authenticator` object even if they were doing entirely un-authed requests!